base/src/commonMain/kotlin/utils/MeasurementUnit.kt

```diff
@@ -0,0 +1,320 @@
+
+private val MILLILITERS_IN_LITER = 1000.0.toDecimal()
+private val UK_FLUID_OUNCES_IN_LITER = 35.1951.toDecimal()
+private val UK_PINTS_IN_LITER = 1.75975.toDecimal()
+private val UK_QUARTS_IN_LITER = 0.879877.toDecimal()
+private val UK_GALLONS_IN_LITER = 0.219969.toDecimal()
+private val US_FLUID_OUNCES_IN_LITER = 33.814.toDecimal()
+private val US_PINTS_IN_LITER = 2.11338.toDecimal()
+private val US_QUARTS_IN_LITER = 1.05669.toDecimal()
+private val US_GALLONS_IN_LITER = 0.264172.toDecimal()
+
+private val KELVIN_FREEZING = 273.15.toDecimal()
+private val FAHRENHEIT_FREEZING = 32.00.toDecimal()
+
+/**
+ * Translates from other systems to base SI
+ *
+ * @param value The non SI value to translate
+ * @param inUnit The type of the input [value] we are translating from
+ */
+private fun toBaseSi(value: Decimal, inUnit: ScientificUnit<*, *>): Decimal {
+    return when (inUnit) {
+        Decimeter -> TODO()
+        Centimeter -> value / CENTIMETERS_IN_METER
+        Millimeter -> value / MILLIMETERS_IN_METER
+        Decameter -> TODO()
+        Hectometer -> TODO()
+        Kilometer -> value / KILOMETERS_IN_METER
+        Meter -> value
+        Inch -> value / INCHES_IN_METER
+        Foot -> value / FEET_IN_METER
+        Yard -> value / YARDS_IN_METER
+        Mile -> value / MILES_IN_METER
+        Celsius -> value + KELVIN_FREEZING
+        Kelvin -> value
+        Fahrenheit -> (value - FAHRENHEIT_FREEZING) * 5.0.toDecimal() / 9.0.toDecimal() + KELVIN_FREEZING
+        Tonne -> value / TONES_IN_KILOGRAM
+        Kilogram -> value
+        Gram -> value / GRAMS_IN_KILOGRAM
+        Milligram -> value / MILLIGRAMS_IN_KILOGRAM
+        Microgram -> TODO()
+        Stone -> value / STONES_IN_KILOGRAM
+        Pound -> value / POUNDS_IN_KILOGRAM
+        Ounce -> value / OUNCES_IN_KILOGRAM
+        Dram -> value / DRAMS_IN_KILOGRAM
+        Grain -> value / GRAINS_IN_KILOGRAM
+        UsTon -> value / US_TONES_IN_KILOGRAM
+        ImperialTon -> value / UK_TONES_IN_KILOGRAM
+        CubicMeter -> TODO()
+        Liter -> value
+        Milliliter -> value / MILLILITERS_IN_LITER
+        UsLiquidGallon -> value / US_GALLONS_IN_LITER
+        UsLiquidQuart -> value / US_QUARTS_IN_LITER
+        UsLiquidPint -> value / US_PINTS_IN_LITER
+        UsLegalCup -> TODO()
+        UsFluidOunce -> value / US_FLUID_OUNCES_IN_LITER
+        UsFluidDram -> TODO()
+        ImperialGallon -> TODO()
+        ImperialQuart -> value / UK_QUARTS_IN_LITER
+        ImperialPint -> value / UK_PINTS_IN_LITER
+        ImperialCup -> TODO()
+        ImperialFluidOunce -> value / UK_FLUID_OUNCES_IN_LITER
+        ImperialFluidDram -> TODO()
+        CubicFoot -> TODO()
+        CubicInch -> TODO()
+    }
+}
+
+/**
+ * Translates from base SI to other systems
+ *
+ * @param value Value in base SI system
+ * @param toUnit The type of the return value
+ */
+private fun fromBaseSi(value: Decimal, toUnit: ScientificUnit<*, *>): Double {
```

Comment:
If I send a decimal I expect to receive one back as well

Reply:
That is an internal private helper fun. It is not meant for public usage. Basically the system works as:

Ounces - ShortTons = Ounces -> Meters -> ShortTones 

I think your point is to expose also from/to SI functionality. I Can do that like:

fun convertToSI(value: Double, from: ScientificUnit.Temperature<*> =
    fromTo(value.toDecimal(), from, to)

fun convertFromSI(value: Double, from: ScientificUnit.Temperature<*> =
    fromTo(value.toDecimal(), from, to)
    
Where I assume for each the input / output will be in SI units.